src/main/java/org/spongepowered/api/command/args/CommandContext.java

```diff
@@ -158,4 +158,29 @@ public boolean hasAny(String key) {
     public boolean hasAny(Text key) {
         return hasAny(ArgUtils.textToArgKey(key));
     }
+
+    /**
+     * Return this arguments object's current state. Can be used to reset with the {@link #setState(Object)} method.
+     *
+     * @return The current state
+     */
+    public Object getState() {
```

Comment:
It's a matter of the intention of the function. I'm mirroring what is done in the `CommandArgs` class, which has methods that get and set objects.

The point of these methods is to capture the state of the object so that it can be reveresed. The consumer of this API does not need to know what it is, only that it holds some state, likewise, they only need to know that they give it back to this object to reset it. By making it an object, we are kind of saying that, "you don't need to know what this, only that it holds the current state." It also means that if changes to this state are made, it's not necessarily an API breaking change.